pkg/resourcemanager/azuresql/azuresqlmanageduser/azuresqlmanageduser.go

```diff
@@ -243,3 +212,15 @@ func findBadChars(stack string) error {
 	}
 	return nil
 }
+
+func makeSecretKey(secretClient secrets.SecretClient, instance *v1alpha1.AzureSQLManagedUser) secrets.SecretKey {
+	if secretClient.GetSecretNamingVersion() == secrets.SecretNamingV1 {
+		if len(instance.Spec.KeyVaultSecretPrefix) != 0 { // If KeyVaultSecretPrefix is specified, use that for secrets
+			return secrets.SecretKey{Name: instance.Spec.KeyVaultSecretPrefix, Namespace: instance.Namespace, Kind: instance.TypeMeta.Kind}
+		}
+		return secrets.SecretKey{Name: instance.Name, Namespace: instance.Namespace, Kind: instance.TypeMeta.Kind}
+	}
+
+	// TODO: Ignoring prefix here... we ok with that?
```

Comment:
Yes. It's also taking an opportunity to remove some of the "extra" secrets configuration that only AzureSQLUser/AzureSQLManagedUser have that don't match the other types and doesn't really seem to add a lot of value to what customers can do (since it's not needed to avoid conflicts anymore it seems just extra complexity on our side to support it without much in the way of customer value add)